### PR TITLE
Consistency in PP checking on use of WOLFSSL_CRYPTO_EX_DATA

### PR DESCRIPTION
**Pull request by kaleb-himes**

### CODE REVIEW

File: wolfssl/internal.h [hunk 1]
```diff
@@ -2777,7 +2777,7 @@ struct WOLFSSL_CTX {
     byte            readAhead;
     void*           userPRFArg; /* passed to prf callback */
 #endif
-#ifdef HAVE_EX_DATA
+#if defined(HAVE_EX_DATA) || defined(FORTRESS)
```
Comment (toddouska) [inline wolfssl/internal.h:2780]: Isn't it simpler to have FORTRESS enable HAVE_EX_DATA in settings if not already on?

Comment (kaleb-himes) [reply wolfssl/internal.h]: Done!

Removed Fortress Conditional and added check in settings.h to set HAVE_EX_DATA if not already defined when FORTRESS is set.

```
#ifdef FORTRESS
     #ifndef HAVE_EX_DATA
         #define HAVE_EX_DATA
     #endif
...
```

**Comment by kaleb-himes** [comment]
retest this please.

**Comment by kaleb-himes** [comment]
retest this please

**Comment by kaleb-himes** [comment]
Super old but still relevant. Updated

**Review by dgarske — APPROVED** [review]

**Pull request merged**